cmd/influxdb-writer/main.go

```diff
@@ -7,174 +7,95 @@ import (
-	"github.com/mainflux/mainflux"
 	"github.com/mainflux/mainflux/consumers"
 	"github.com/mainflux/mainflux/consumers/writers/api"
 	"github.com/mainflux/mainflux/consumers/writers/influxdb"
+	"github.com/mainflux/mainflux/internal"
+	influxClient "github.com/mainflux/mainflux/internal/client/influxdb"
+	"github.com/mainflux/mainflux/internal/env"
+	"github.com/mainflux/mainflux/internal/server"
+	httpserver "github.com/mainflux/mainflux/internal/server/http"
 	"github.com/mainflux/mainflux/logger"
-	"github.com/mainflux/mainflux/pkg/errors"
 	"github.com/mainflux/mainflux/pkg/messaging/brokers"
-	stdprometheus "github.com/prometheus/client_golang/prometheus"
 	"golang.org/x/sync/errgroup"
 )
 
 const (
-	svcName      = "influxdb-writer"
-	stopWaitTime = 5 * time.Second
-
-	defBrokerURL  = "nats://localhost:4222"
-	defLogLevel   = "error"
-	defPort       = "8180"
-	defDB         = "mainflux"
-	defDBHost     = "localhost"
-	defDBPort     = "8086"
-	defDBUser     = "mainflux"
-	defDBPass     = "mainflux"
-	defConfigPath = "/config.toml"
-
-	envBrokerURL  = "MF_BROKER_URL"
-	envLogLevel   = "MF_INFLUX_WRITER_LOG_LEVEL"
-	envPort       = "MF_INFLUX_WRITER_PORT"
-	envDB         = "MF_INFLUXDB_DB"
-	envDBHost     = "MF_INFLUXDB_HOST"
-	envDBPort     = "MF_INFLUXDB_PORT"
-	envDBUser     = "MF_INFLUXDB_ADMIN_USER"
-	envDBPass     = "MF_INFLUXDB_ADMIN_PASSWORD"
-	envConfigPath = "MF_INFLUX_WRITER_CONFIG_PATH"
+	svcName           = "influxdb-writer"
+	envPrefix         = "MF_INFLUX_WRITER_"
+	envPrefixHttp     = "MF_INFLUX_WRITER_HTTP_"
+	envPrefixInfluxdb = "MF_INFLUXDB_"
+	defSvcHttpPort    = "8180"
 )
 
 type config struct {
-	brokerURL  string
-	logLevel   string
-	port       string
-	dbName     string
-	dbHost     string
-	dbPort     string
-	dbUser     string
-	dbPass     string
-	configPath string
+	LogLevel   string `env:"MF_INFLUX_WRITER_LOG_LEVEL"     envDefault:"info"`
+	ConfigPath string `env:"MF_INFLUX_WRITER_CONFIG_PATH"   envDefault:"/config.toml"`
+	BrokerURL  string `env:"MF_BROKER_URL"                  envDefault:"nats://localhost:4222"`
 }
 
 func main() {
-	cfg, clientCfg := loadConfigs()
 	ctx, cancel := context.WithCancel(context.Background())
 	g, ctx := errgroup.WithContext(ctx)
 
-	logger, err := logger.New(os.Stdout, cfg.logLevel)
+	cfg := config{}
+	if err := env.Parse(&cfg); err != nil {
+		log.Fatalf("failed to load %s configuration : %s", svcName, err.Error())
+	}
+
+	logger, err := logger.New(os.Stdout, cfg.LogLevel)
 	if err != nil {
 		log.Fatalf(err.Error())
 	}
 
-	pubSub, err := brokers.NewPubSub(cfg.brokerURL, "", logger)
+	pubSub, err := brokers.NewPubSub(cfg.BrokerURL, "", logger)
 	if err != nil {
-		logger.Error(fmt.Sprintf("Failed to connect to message broker: %s", err))
-		os.Exit(1)
+		log.Fatalf("failed to connect to message broker: %s", err.Error())
 	}
 	defer pubSub.Close()
 
-	client, err := influxdata.NewHTTPClient(clientCfg)
+	influxdbConfig := influxClient.Config{}
```

Comment:
Maybe `influxConfig`, or `influxDBConfig`

Reply:
Change all `influxdb` and `influx` to `influxDB`